### PR TITLE
Add remaining indices from intial scoping with John

### LINKED ISSUE #4: add remaining 9 indices initially discussed with John W to pipeline
The main event of the current pipeline is the production of an annual dataset of indices that is used for generation of other summary products. Currently that dataset consists of four indices: rx1day, hsd, hd, and cd, identified as priorities when scoping the project. 

There are 9 remaining from this original set of those identified by John W during scoping - with no urgent need for other summary products at the moment, it's time to add these!

* SU: Summer Days – Annual number of days with Tmax above 25 C
* DW: Deep Winter days – Annual number of days with Tmin below -30 C
* WSDI: Warm Spell Duration Index -- Annual count of occurrences of at least 5 consecutive days with daily mean T above 90 th percentile of historical values for the date
* CDSI: Cold Spell Duration Index -- Same as WDSI, but for daily mean T below 10 th percentile
* Rx5day – Annual maximum 5-day precipitation
* R10mm: Number of heavy precip days – Annual count of days with precip > 10 mm
* CWD: Consecutive wet days – Yearly number of the most consecutive days with precip > 1 mm
* CDD: Consecutive dry days – Same as CED, but for days with precip < 1 mm
* WndD: Windy Days – Yearly number of days with mean wind speed > 10 m/sec

### LINKED ISSUE #5: Change heavy snow days index to mean of 5 snowiest snows
John has requested we change the algorithm for the heavy snow days index to be the mean of the 5 largest snowfall periods for each year.

### PR DESCRIPTION
**Pull request by kyleredilla**
This PR contains processing code used for generating the remaining batch of climate indices from the bias-corrected CORDEX data provided by collaborators on the Two Bears Environmental Consulting project.

To test, follow the README and see if you can run the main pipeline (`process_indices.ipynb`) which produces the annual dataset of indices for the whole grid. Try creating a conda env from the env file instead of using an existing one to make sure it works with the processing code. Then execute all of the `extract_indices.ipynb` and `location_summary_plots.ipynb` notebooks to generate point-based extractions and bar plots, respectively. Also run the `qc.ipynb` notebook to spot check the new indices dataset and point extrations.

I would recommend setting the `CORDEX_DIR` variable to `/atlas_scratch/kmredilla/CORDEX_AK`, instead of the version of the dataset on Poseidon. The current notebook might not even work with without doing that, it hasn't been tested. Things were very slow / boggy but performance has been excellent after copying the dataset to scratch space on Atlas. Also probably best to set the `OUTPUT_DIR` variable to be on your own scratch space, so existing products are not overwritten. 

closes #4 
closes #5

### CODE REVIEW

File: process_indices.ipynb [hunk 1]
```diff
@@ -135,8 +149,19 @@
     "    #  pool-ing to then construct new Dataset\n",
     "    \n",
     "    with xr.open_dataset(fp) as ds:\n",
-    "        out = [indices.compute_index(ds[varname], index, model, scenario) for index in index_list]\n",
-    "    \n",
+    "        # out = [indices.compute_index(ds[varname], index, model, scenario) if index not in [\"wsdi\", \"csdi\"] else indices.compute_index(ds[varname], index, model, scenario) for index in index_list]\n",
```
Comment (cstephen) [inline process_indices.ipynb]: Can this commented out code be removed? If not, I'd suggest adding a comment above it to indicate why/when it should be uncommented.

File: location_summary_plots.ipynb [hunk 2]
```diff
@@ -270,9 +75,9 @@
     "        .agg({aggr_var: [\"min\", \"mean\", \"max\"]})\n",
     "    )\n",
     "    \n",
-    "    if len(np.unique(temp_df[\"mean\"])) < 5:\n",
-    "        print(f\"No meaningful variation, skipping location: {' '.join(title.split()[-2:])}, var: {index}\")\n",
-    "        return \"skip\"\n",
+    "    # if len(np.unique(temp_df[aggr_var])) < 5:\n",
+    "    #     print(f\"No meaningful variation, skipping location: {' '.join(title.split()[-2:])}, var: {index}\")\n",
+    "    #     return \"skip\"\n",
```
Comment (cstephen) [inline location_summary_plots.ipynb]: Same as my other comment re: commented out code. Can this commented out code be removed? If not, I'd suggest adding a comment above it to indicate why/when it should be uncommented.

**Review by cstephen — APPROVED** [review]
Everything works for me. Great work, and lots of it!

Some review notes:

- I created a fresh conda environment using environment.yml and didn't run into any environment hassles.
- I ran each notebook mentioned in the PR description with no issues.
- I spot checked the summary plot files for a couple locations for each variable, and they looked good & made sense.

I added a couple code comments regarding commented out code. No showstoppers, though, so I'm marking this as approved.

**Comment by charparr** [comment]
I spun up a fresh env with `environment.yml` and ran all the suggested notebooks with zero hiccups - nice work! It is great to see `xclim` being wielded so effectively ⚔️ 
I'm marking this as approved and leave it for @kyleredilla to actually merge in case he wants to remove (or retain) the commented out snippets that Craig mentioned.

**Review by charparr — APPROVED** [review]

File: location_summary_plots.ipynb (hunk 2 above)
Comment (kyleredilla) [reply location_summary_plots.ipynb]: Thanks for the catches!

**Pull request merged**